airbyte-integrations/connectors/source-instagram/source_instagram/streams.py

```diff
@@ -218,8 +238,16 @@ def read_records(
                 f"temporarily unavailable data."
             )
             self.should_exit_gracefully = True
+
         yield from complete_records
 
+        # update state using IncrementalMixin
+        # reference issue: https://github.com/airbytehq/airbyte/issues/24697
+        if sync_mode == SyncMode.incremental:
+            if complete_records:
```

Comment:
better to use `AND` to increase if statements qty

Reply:
Done